### PR TITLE
FACT-2444 add Debian legacy facts

### PR DESCRIPTION
**Pull request by Filipovici-Andrei**

### CODE REVIEW

File: lib/facts/debian/memory/swap/available_bytes.rb [hunk 1]
```diff
@@ -6,10 +6,13 @@ module Memory
...
         class AvailableBytes
           FACT_NAME = 'memory.swap.available_bytes'
+          ALIASES = 'swapfree_mb'
 
           def call_the_resolver
             fact_value = Facter::Resolvers::Linux::Memory.resolve(:swap_free)
-            Facter::ResolvedFact.new(FACT_NAME, fact_value)
+
+            [Facter::ResolvedFact.new(FACT_NAME, fact_value),
+             Facter::ResolvedFact.new(ALIASES, fact_value ? (fact_value / (1024.0 * 1024.0)).round(2) : nil, :legacy)]
```
Comment (sebastian-miclea) [inline lib/facts/debian/memory/swap/available_bytes.rb]: Could you please move the logic outside the method call?

File: lib/facts/debian/memory/swap/total_bytes.rb [hunk 2]
```diff
@@ -6,10 +6,12 @@ module Memory
       module Swap
         class TotalBytes
           FACT_NAME = 'memory.swap.total_bytes'
+          ALIASES = 'swapsize_mb'
 
           def call_the_resolver
             fact_value = Facter::Resolvers::Linux::Memory.resolve(:swap_total)
-            Facter::ResolvedFact.new(FACT_NAME, fact_value)
+            [Facter::ResolvedFact.new(FACT_NAME, fact_value),
+             Facter::ResolvedFact.new(ALIASES, fact_value ? (fact_value / (1024.0 * 1024.0)).round(2) : nil, :legacy)]
```
Comment (sebastian-miclea) [inline lib/facts/debian/memory/swap/total_bytes.rb]: Could you please move the logic outside the method call?

File: lib/facts/debian/memory/system/available_bytes.rb [hunk 3]
```diff
@@ -6,10 +6,13 @@ module Memory
...
         class AvailableBytes
           FACT_NAME = 'memory.system.available_bytes'
+          ALIASES = 'memoryfree_mb'
 
           def call_the_resolver
             fact_value = Facter::Resolvers::Linux::Memory.resolve(:memfree)
-            Facter::ResolvedFact.new(FACT_NAME, fact_value)
+
+            [Facter::ResolvedFact.new(FACT_NAME, fact_value),
+             Facter::ResolvedFact.new(ALIASES, fact_value ? (fact_value / (1024.0 * 1024.0)).round(2) : nil, :legacy)]
```
Comment (sebastian-miclea) [inline lib/facts/debian/memory/system/available_bytes.rb]: Could you please move the logic outside the method call?

File: lib/facts/debian/os/selinux.rb [hunk 4]
```diff
@@ -5,17 +5,12 @@ module Debian
...
           selinux = Facter::Resolvers::SELinux.resolve(:enabled)
 
-          Facter::ResolvedFact.new(FACT_NAME, build_fact_list(selinux))
-        end
-
-        def build_fact_list(selinux)
-          {
-            enabled: selinux
-          }
+          [Facter::ResolvedFact.new(FACT_NAME, enabled: selinux), Facter::ResolvedFact.new(ALIASES, selinux, :legacy)]
```
Comment (sebastian-miclea) [inline lib/facts/debian/os/selinux.rb]: Could you please move the logic outside the method call?

File: lib/facts/debian/processors/isa.rb [hunk 5]
```diff
@@ -5,11 +5,13 @@ module Debian
     module Processors
       class Isa
         FACT_NAME = 'processors.isa'
+        ALIASES = 'hardwareisa'
 
         def call_the_resolver
           fact_value = Facter::Resolvers::Uname.resolve(:processor)
 
-          Facter::ResolvedFact.new(FACT_NAME, get_isa(fact_value))
+          [Facter::ResolvedFact.new(FACT_NAME, get_isa(fact_value)),
```
Comment (sebastian-miclea) [inline lib/facts/debian/processors/isa.rb]: You could save the value into a var and not process it twice.

File: lib/facts/debian/memory/swap/available_bytes.rb [hunk 6]
```diff
@@ -6,10 +6,19 @@ module Memory
...
+
+            [Facter::ResolvedFact.new(FACT_NAME, fact_value),
+             Facter::ResolvedFact.new(ALIASES, convert_to_mb(fact_value), :legacy)]
+          end
+
+          private
+
+          def convert_to_mb(fact_value)
+            fact_value ? (fact_value / (1024.0 * 1024.0)).round(2) : nil
           end
```
Comment (gimmyxd) [inline lib/facts/debian/memory/swap/available_bytes.rb]: this method should be extracted in separate file as it is duplicated in 4 places.

**Review by sebastian-miclea — APPROVED** [review]

**Pull request merged**